### PR TITLE
Adds script to create HTML test coverage report

### PR DESCRIPTION
**Pull request by FangedParakeet**
Fixes nothing. This is purely for developer benefit, so that we can more accurately analyse where our coverage is lacking and identify areas to improve coverage when we dip beneath our coverage floor of 60%.

Please note that the coverage percentage that is checked against for commits uses [Clover](https://openclover.org/doc/manual/4.2.0/hacking--using-openclover-for-php.html), whereas this HTML report uses [phpunit's native coverage checker](https://docs.phpunit.de/en/9.5/code-coverage-analysis.html). [These are not exactly the same](https://stackoverflow.com/questions/34296485/how-is-coverage-html-different-from-coverage-clover), so the percentages calculated will likely differ marginally. However, generating a similar HTML report with Clover is non-trivial and requires additional add-ons, so using phpunit's native tool is a lightweight approximation that is very, very close to the more accurate coverage percentage produced by Clover and is still very useful at helping identifying areas of lapsing coverage. 

#### Changes proposed in this Pull Request

Adds new command `npm run test:php-coverage` that will generate a test coverage report at `http://localhost:8082/php-test-coverage`.

#### Testing instructions

1. Install test packages and bring up Docker containers. `nvm use && npm i && npm run up`.
2. Run `npm run test:php-coverage`. After PHP tests have finished running, phpunit will create a test coverage report inside your local Docker directory (`docker/wordpress/php-test-coverage`).
3. Visit your local site URL at `http://localhost:8082/php-test-coverage` to view a complete report of test coverage for all relevant PHP files in the repo.

-------------------

- [x] Run `npm run changelog` to add a changelog file, choose `patch` to leave it empty if the change is not significant. You can add multiple changelog files in one PR by running this command a few times. 
- [x] Covered with tests (or have a good reason not to test in description ☝️)
- [x] Tested on mobile (or does not apply)

**Post merge**

- [ ] Link to testing instructions from [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) following [these instructions](https://github.com/Automattic/woocommerce-payments/wiki/How-to-write-good-manual-testing-scenarios) : _Add link here / 'QA Testing Not Applicable'_
- [ ] Add or update [critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Critical-flows) and [testing instructions for critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Testing-instructions-for-critical-flows), if applicable.
- [ ] Add what's changed (description, screenshot, demo videos etc.) to the release announcement post, if applicable.

### CODE REVIEW

**Comment by botwoo** [comment]
### Test the build 

#### Option 1. Jetpack Beta

- Install and activate [Jetpack Beta](https://jetpack.com/download-jetpack-beta/).
- Use this build by searching for PR number `5838` or branch name `test/check-code-coverage` in your-test.site/wp-admin/admin.php?page=jetpack-beta&plugin=woocommerce-payments          

#### Option 2. Jurassic Ninja - available for logged-in A12s

:rocket: [Launch a JN site with this branch](https://jurassic.ninja/create/?jetpack-beta&shortlived&nojetpack&woocommerce&branches.woocommerce-payments=test/check-code-coverage) :rocket:

:information_source: Install this [Tampermonkey script](https://github.com/Automattic/woocommerce-payments/tree/develop/bin/wcpay-live-branches) to get more options.

---

Build info: 

- Latest commit: 91b538f8daa8afd99e9da331fd730dcc0d5f0a66
- Build time: 2023-05-10 15:18:29 UTC

_Note: the build is updated when a new commit is pushed to this PR._

File: bin/check-test-coverage.sh [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+while getopts ':w' OPTION; do
+	case $OPTION in
+		w)
+		WATCH_FLAG=true
+		shift
+		;;
+	esac
+done
+
```
Comment (mdmoore) [inline bin/check-test-coverage.sh]: Is this flag needed? I'm trying to think a use case that would involve running watcher when generating a coverage report. It's possible that there are good reasons to leave this that I'm just missing.

**Review by mdmoore — APPROVED** [review]
This is working great! I just left one small question. The report is quite nice and a great addition to the local dev environment. Thanks for working on this!

File: bin/check-test-coverage.sh (hunk 1 above)
Comment (FangedParakeet) [reply bin/check-test-coverage.sh]: Oof, nice catch. I don't use this flag at all. Amended in d9dcb5a938c7f2f89f6e3933ae921fec064af07b. Cheers! 👍

**Review by mdmoore — APPROVED** [review]

**Pull request merged**